src/services/cw721/cw721-reindexing.service.ts

```diff
@@ -0,0 +1,158 @@
+    super(broker);
+  }
+
+  @QueueHandler({
+    queueName: BULL_JOB_NAME.REINDEX_CW721_CONTRACT,
+    jobName: BULL_JOB_NAME.REINDEX_CW721_CONTRACT,
+  })
+  async jobHandler(_payload: ICw721ReindexingServiceParams): Promise<void> {
+    const { smartContractId, contractAddress } = _payload;
+    const cw721Contract = await CW721Contract.query()
+      .withGraphJoined('smart_contract')
+      .where('smart_contract.address', contractAddress)
+      .select(['cw721_contract.id'])
+      .first();
+    // query
+    const contractInfo = (
+      await CW721Contract.getContractsInfo([contractAddress])
+    )[0];
+    const currentTokensOwner = await CW721Contract.getAllTokensOwner(
+      contractAddress
+    );
+    const minUpdatedHeightOwner =
+      _.min(
+        currentTokensOwner.map((tokenOwner) => tokenOwner.last_updated_height)
+      ) || 0;
+    const maxUpdatedHeightOwner =
+      _.max(
+        currentTokensOwner.map((tokenOwner) => tokenOwner.last_updated_height)
+      ) || 0;
+    if (cw721Contract) {
+      await CW721Activity.query()
+        .delete()
+        .where('cw721_contract_id', cw721Contract.id);
+      await CW721ContractStats.query()
+        .delete()
+        .where('cw721_contract_id', cw721Contract.id);
+      await CW721Token.query()
+        .delete()
+        .where('cw721_contract_id', cw721Contract.id);
+      await CW721Contract.query().deleteById(cw721Contract.id);
+    }
+    await CW721Contract.query().insertGraph({
+      ...CW721Contract.fromJson({
+        contract_id: smartContractId,
+        symbol: contractInfo?.symbol,
+        minter: contractInfo?.minter,
+        name: contractInfo?.name,
+        track: true,
+      }),
+      tokens: currentTokensOwner.map((tokenOwner) =>
+        CW721Token.fromJson({
+          token_id: tokenOwner.token_id,
+          media_info: null,
+          owner: tokenOwner.owner,
+          last_updated_height: tokenOwner.last_updated_height,
+          burned: false,
+        })
+      ),
+    });
+    // handle from minUpdatedHeightOwner to blockHeight
+    await this.broker.call(
+      SERVICE.V1.Cw721.HandleRangeBlockMissingContract.path,
+      {
+        smartContractId,
+        startBlock: minUpdatedHeightOwner,
+        endBlock: maxUpdatedHeightOwner,
+      } satisfies IContextReindexingServiceHistory
+    );
+    // insert histories
+    await this.createJob(
+      BULL_JOB_NAME.REINDEX_CW721_HISTORY,
+      BULL_JOB_NAME.REINDEX_CW721_HISTORY,
+      {
+        smartContractId,
+        startBlock: config.crawlBlock.startBlock,
+        endBlock: maxUpdatedHeightOwner,
+        currentId: 0,
+        contractAddress,
+      } satisfies ICw721ReindexingHistoryParams,
+      {
+        removeOnComplete: true,
+      }
+    );
+  }
+
+  @Action({
+    name: SERVICE.V1.CW721ReindexingService.ReindexingApi.key,
+    params: {
+      contractAddress: 'string',
+    },
+  })
+  public async reindexingApi(ctx: Context<IAddressParam>) {
```

Comment:
action thì bỏ chữ api đi,
toàn bộ cái tên đang là 
`cw721ReindexingService.reindexingApi` 

để là `cw721ReindexingService.reindex` là thừa rồi